packages/node/src/dev-server.ts

```diff
@@ -228,24 +229,33 @@ async function compileUserCode(
   }
 }
 
-async function createEdgeRuntime(userCode: string | undefined) {
+async function createEdgeRuntime(params?: {
+  userCode: string;
+  wasmAssets: WasmAssets;
+}) {
   try {
-    if (!userCode) {
+    if (!params) {
       return undefined;
     }
 
+    const wasmBindings = await params.wasmAssets.getContext();
     const edgeRuntime = new EdgeRuntime({
-      initialCode: userCode,
+      initialCode: params.userCode,
       extend: (context: Primitives) => {
-        Object.assign(context, {
-          __dirname: '',
-          module: {
-            exports: {},
-          },
-          process: {
-            env: process.env,
+        Object.assign(
+          context,
+          {
+            __dirname: '',
```

Comment:
Do you also need `__filename` for consistency?

Reply:
I was only adding what was necessary to support the wrapping logic used by esbuild to bundle the code into a single file. If we add too much here, it may hide the actual errors from the user code because `__filename` will exist, but it won't have an expected value, causing some other error.

I should have added a comment about what exactly this was doing.

It's good that you brought this up! These values were necessary for the `ncc` bundle, but after switching to `esbuild`, I didn't re-evaluate that they were all still required. It turns out they are not.

I updated this with less values and comments about why they are here.

Reply:
This change isn't strictly related to this PR, though. I opened a new PR with these changes: https://github.com/vercel/vercel/pull/8260